### PR TITLE
Added step 2 implementation

### PR DESCRIPTION
**Pull request by DariaWelt**

### CODE REVIEW

File: intelligent_placer_lib/detection.py [hunk 1]
```diff
@@ -1,101 +1,132 @@
...
+    if not is_red:
+        return False
+    cur_res = cv2.matchShapes(object_contour, circle_contour, cv2.cv2.CONTOURS_MATCH_I3, 0.0)
+    return cur_res < 0.1
 
-def _classify_objects(contours, image: np.ndarray, verbose: bool = False) -> Tuple[np.ndarray, List]:
+
+def is_blue_rectangle(object_mask: np.ndarray, object_image: np.ndarray, object_contour) -> bool:
+    channels_av = get_channels_average(object_mask, object_image)
+    is_blue = not np.isnan(channels_av[2]) and channels_av[2] > 0.6 and channels_av[0] < 0.2 and channels_av[1] < 0.6
```
Comment (OvechkinDanil) [inline intelligent_placer_lib/detection.py:61]: Should these constants be in the code? Maybe create variables for them. 
Check the same in all methods

File: intelligent_placer_lib/detection.py [hunk 2]
```diff
@@ -1,101 +1,132 @@
...
+    cv2.drawContours(cur_segmented, [contour], -1, object_class, cv2.FILLED)
+    cv2.drawContours(segmentation, [contour], -1, object_class, cv2.FILLED)
+    if verbose:
+        plt.imshow(cur_segmented + 220)
+        plt.title(f'segmented class {object_class}')
+        plt.show()
+    return segmentation
+
+
+def _classify_objects(contours, polygon_max_x: int, image: np.ndarray, verbose: bool = False) -> np.ndarray:
```
Comment (OvechkinDanil) [inline intelligent_placer_lib/detection.py:80]: I'm not a good in writing python code, but as for me It is better to use blank lines in order to separate logical blocks of code

File: intelligent_placer_lib/detection.py [hunk 3]
```diff
@@ -1,101 +1,132 @@
...
+                                 (np.linalg.norm(item_des[match[0]]) * np.linalg.norm(im_des[match[1]]))
+                                 for match in pairs]
+                    cur_matched = len(list(filter(lambda elem: elem < 0.002, distances)))
+                    if cur_matched > best_matched:
+                        best_matched = cur_matched
+                        res = cur_res
+                        matched_class = item_class
+            if res == 1:
+                continue
+            if best_matched < 10:
```
Comment (OvechkinDanil) [inline intelligent_placer_lib/detection.py:117]: Might be worth combining these two conditions into one

File: README.md [hunk 4]
```diff
@@ -75,4 +76,19 @@ for testing:
...
+- added exceptional check for red circle and blue phone(rectangle)
+- resized input images so algorithm works faster
+- started implementation of genetic algo for placing problem
+- extended dataset (added more examples with rotations and included example with shadow impact where algorithm don't work)
+
+## Improvements todo from version 2
+- add the ability to parallelize evaluations in brute force parallel-rotate solution
+- implement genetic algo
+- add exceptional check for keychain pass
+- evaluate precicion, accuracy and recall automatically (only manual for now)
```
Comment (derkanw) [inline README.md:94]: I think, it is also necessary to describe the metrics specified in the .ipynb-file here

File: README.md [hunk 5]
```diff
@@ -32,6 +32,15 @@ https://drive.google.com/drive/folders/1gk90RoNKPqP_lwWpY8cYFQpLThSHtY2m?usp=sha
...
+## Get started
+- clone repository
+- upload photos of possible items with masks from drive and put it into `intelligent_placer_lib/data/`. If you want to 
+use yor own data, you could put it (images and masks) into this dir.
+
+for testing:
+- put dataset into `test/data`
+- put dataset specification json into `test/data`. Example of such file already put here.
+
 ## Placer Algorithm
```
Comment (derkanw) [inline README.md:44]: It seems, that it is necessary to describe the final version of the algorithm. And the process of its previous changes can be traced by improvement points

File: intelligent_placer_lib/detection.py [hunk 6]
```diff
@@ -1,101 +1,132 @@
...
+    mask = np.zeros(shape)
+    return cv2.drawContours(mask, [contour], -1, 255, cv2.FILLED)
+
+
+def get_channels_average(object_mask: np.ndarray, object_image: np.ndarray) -> np.ndarray:
+    object_part = np.array([object_image[..., i] * (object_mask / 255).astype('uint8') for i in range(object_image.shape[-1])])
+    return np.sum(np.sum(object_part, axis=1), axis=1) / np.sum(object_mask > 0) / 255
+
 
+def is_red_circle(object_mask: np.ndarray, object_image: np.ndarray, object_contour, circle_contour) -> bool:
```
Comment (derkanw) [inline intelligent_placer_lib/detection.py:50]: I did not understand, why the classes of the red circle and the blue rectangle are considered separately?

File: intelligent_placer_lib/placer.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+from typing import List, Tuple
+
+import matplotlib.pyplot as plt
+import cv2
+import numpy as np
+from skimage.measure import regionprops
+
+
+def estimate_possibility(figure_mask: np.ndarray, objects_masks: List[np.ndarray]) -> bool:
+    if np.sum(figure_mask > 0) < sum([np.sum(object_mask > 0) for object_mask in objects_masks]):
```
Comment (derkanw) [inline intelligent_placer_lib/placer.py:11]: You can return the expression standing in if

**Pull request merged**